ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/TlsHttpClient.java

```diff
@@ -0,0 +1,176 @@
+      trustManagerFactory.init(trustStore);
+      this.trustManagerFactory = trustManagerFactory;
+    } catch (GeneralSecurityException gse) {
+      throw new RuntimeException("Unable to load trust manager factory", gse);
+    }
+  }
+
+  private void initKeyManagerFactory() {
+    if (!isClientAuthRequired()) {
+      return;
+    }
+
+    try {
+      final char[] password = getKeystorePassword(keyStorePasswordResource);
+      final KeyStore keyStore = loadP12KeyStore(keyStoreResource, password);
+      final KeyManagerFactory keyManagerFactory = KeyManagerFactory.getInstance("PKIX");
+      keyManagerFactory.init(keyStore, password);
+      this.keyManagerFactory = keyManagerFactory;
+    } catch (GeneralSecurityException gse) {
+      throw new RuntimeException("Unable to load key manager factory", gse);
+    }
+  }
+
+  private char[] getKeystorePassword(final String passwordResource) {
+    try {
+      final File passwordFile =
+          Paths.get(ClassLoader.getSystemResource(passwordResource).toURI()).toFile();
+
+      return Files.asCharSource(passwordFile, Charsets.UTF_8).readFirstLine().toCharArray();
+    } catch (URISyntaxException | IOException e) {
+      throw new RuntimeException("Unable to read keystore password file", e);
+    }
+  }
+
+  private KeyStore loadP12KeyStore(final String resource, final char[] password)
+      throws KeyStoreException, NoSuchAlgorithmException, CertificateException {
+    final KeyStore store = KeyStore.getInstance("pkcs12");
+    try (final InputStream keystoreStream = ClassLoader.getSystemResource(resource).openStream()) {
+      store.load(keystoreStream, password);
+    } catch (IOException e) {
+      throw new RuntimeException("Unable to load keystore.", e);
+    }
+    return store;
+  }
+
+  public static final class TlsHttpClientBuilder {
+    private String keyStoreResource;
+    private String keyStorePasswordResource;
+    private String trustStoreResource;
+    private String trustStorePasswordResource;
+
+    private TlsHttpClientBuilder() {}
+
+    public static TlsHttpClientBuilder aTlsHttpClient() {
+      return new TlsHttpClientBuilder();
+    }
+
+    public TlsHttpClientBuilder withKeyStoreResource(String keyStoreResource) {
+      this.keyStoreResource = keyStoreResource;
+      return this;
+    }
+
+    public TlsHttpClientBuilder withKeyStorePasswordResource(String keyStorePasswordResource) {
+      this.keyStorePasswordResource = keyStorePasswordResource;
+      return this;
+    }
+
+    public TlsHttpClientBuilder withTrustStoreResource(String trustStoreResource) {
+      this.trustStoreResource = trustStoreResource;
+      return this;
+    }
+
+    public TlsHttpClientBuilder withTrustStorePasswordResource(String trustStorePasswordResource) {
+      this.trustStorePasswordResource = trustStorePasswordResource;
+      return this;
+    }
+
+    public TlsHttpClient build() {
+      TlsHttpClient tlsHttpClient = new TlsHttpClient();
```

Comment:
nit: Generally prefer to see the builder push all params in via the object's constructor (and making them final) rather than setting the variables directly.